backend/src/controllers/notes.ts

```diff
@@ -140,10 +170,105 @@ export const deleteNote: RequestHandler = async (req, res, next) => {
       throw createHttpError(401, "You cannot access this note");
     }
 
-    await note.remove();
+    note.deletedAt = new Date();
+
+    await note.save();
 
     res.sendStatus(204);
   } catch (error) {
     next(error);
   }
 };
+
+export const getDeletedNotes: RequestHandler = async (req, res, next) => {
+  const authenticatedUserId = req.session.userId;
+
+  try {
+    assertIsDefined(authenticatedUserId);
+
+    const notes = await NoteModel.find({
+      userId: authenticatedUserId,
+      deletedAt: { $ne: null },
+    }).exec();
+    res.status(200).json(notes);
```

Comment:
what if this is null?

Reply:
It's for filtering. If deletedAt field is null, it means It's not deleted.